src/main/java/seedu/address/storage/JsonAdaptedItem.java

```diff
@@ -27,20 +27,24 @@
     private final String quantity;
     private final String supplier;
     private final List<JsonAdaptedTag> tagged = new ArrayList<>();
+    private final String maxQuantity;
 
     /**
      * Constructs a {@code JsonAdaptedItem} with the given item details.
      */
     @JsonCreator
-    public JsonAdaptedItem(@JsonProperty("name") String name, @JsonProperty("quantity") String quantity,
+    public JsonAdaptedItem(@JsonProperty("name") String name,
+                           @JsonProperty("quantity") String quantity,
                            @JsonProperty("supplier") String supplier,
-                           @JsonProperty("tagged") List<JsonAdaptedTag> tagged) {
+                           @JsonProperty("tagged") List<JsonAdaptedTag> tagged,
+                           @JsonProperty("maxQuantity") String maxQuantity) {
         this.name = name;
         this.quantity = quantity;
         this.supplier = supplier;
         if (tagged != null) {
             this.tagged.addAll(tagged);
         }
+        this.maxQuantity = maxQuantity == null ? "0" : maxQuantity;
```

Comment:
Perhaps initializing maxQuantity to a value other than "0" is a better idea? Similarly to tags maybe leaving it empty is fine